common/src/main/java/org/corfudb/common/CorfuSourceCodeVersion.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.common;
+
+
+/**
+ * Static class for representing the corfu source code version using git commit id.
+ */
+public final class CorfuSourceCodeVersion {
+
+    /**
+     * Original git commit id of the head of master branch, which should be updated upon every
+     * corfu artifact deployment.
+     */
+    private static final String CORFU_SOURCE_VERSION_STRING = "d42d17b360acfd2eccac4862b8f1fd1a79f75586";
+
+    /**
+     * Cutoff of the original git commit id to uniquely identify a commit.
+     */
+    private static final int CUTOFF = 12;
+
+    /**
+     * Signals the parser that the git commit is hexadecimal.
+     */
+    private static final int HEX_OPT = 16;
+
+    /**
+     * Getter for the long value which will be populated into our protobuf header. We only take the
+     * first {@link #CUTOFF} characters of {@link #CORFU_SOURCE_VERSION_STRING} as they
+     * are more than enough to uniquely identify a commit in our codebase.
+     *
+     * @return Long value converted from the first 12 characters of {@link #CORFU_SOURCE_VERSION_STRING}
+     */
+    public static long getCorfuSourceCodeVersion() {
+        return Long.parseLong(CORFU_SOURCE_VERSION_STRING.substring(0, CUTOFF), HEX_OPT);
```

Comment:
GitRepositoryState should be able to provide all needed information, please check this out https://github.com/CorfuDB/CorfuDB/blob/d42d17b360acfd2eccac4862b8f1fd1a79f75586/infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java#L433

Reply:
Yeah we also explored that option, but we want a separate static definition because instead of updating this "version" every time the master branch get updated, we may just do it for every release or artifact deployment. This PR is still in draft mode and we are continue exploring automatic way to update the commit id here like using scripting, and it will be populated to other places like the handshake and layout.

Reply:
GitRepositoryState isn't a library, it is a class written to extract certain generated files like "git.branch" etc and then populating it. These property files need to be correctly filled up for it to work.
What @xcchang is doing is essentially short-circuiting the process and making it directly available so it works in all the cases.
Automation to fill this can be added later.

Reply:
BTW, can we get corfu version from corfu app? Do we need anything else than the version? 
I mean we have "0.3.2.1" we know everything about the build, and release number?

I'm trying to understand the motivation here, it seems that ($major.$minor.$fix) semver works well in our case?

Reply:
good question - it's just a safeguard to help folks with build to build breakages, if we retain major.minor then it becomes hard to test build-to-build upgrade failures, since these versions are not going to be modified in build-to-build setups. 
also having a git commit id in the version field protects us from the case where the version wasn't updated.

Reply:
Note: We had off-line discussions and found the `git.properties` file is actually carried into `runtime.jar` and `infrastructure.jar`, so it could be good to use the `GitRepositoryState`.